flake.nix

```diff
@@ -2,115 +2,120 @@
   description = "Compositor for the COSMIC desktop environment";
 
   inputs = {
-    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    # <https://github.com/nix-systems/default-linux>
+    systems.url = "github:nix-systems/default-linux";
 
-    parts.url = "github:hercules-ci/flake-parts";
-    parts.inputs.nixpkgs-lib.follows = "nixpkgs";
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
 
     crane.url = "github:ipetkov/crane";
 
-    rust.url = "github:oxalica/rust-overlay";
-    rust.inputs.nixpkgs.follows = "nixpkgs";
-
-    nix-filter.url = "github:numtide/nix-filter";
+    rust-overlay.url = "github:oxalica/rust-overlay";
+    rust-overlay.inputs.nixpkgs.follows = "nixpkgs";
   };
 
-  outputs =
-    inputs@{
-      self,
-      nixpkgs,
-      parts,
-      crane,
-      rust,
-      nix-filter,
-      ...
-    }:
-    parts.lib.mkFlake { inherit inputs; } {
-      systems = [
-        "aarch64-linux"
-        "x86_64-linux"
-      ];
-
-      perSystem =
-        {
-          self',
-          lib,
-          system,
-          ...
-        }:
-        let
-          pkgs = nixpkgs.legacyPackages.${system}.extend rust.overlays.default;
-          rust-toolchain = pkgs.rust-bin.fromRustupToolchainFile ./rust-toolchain.toml;
-          craneLib = (crane.mkLib pkgs).overrideToolchain rust-toolchain;
-          craneArgs = {
-            pname = "cosmic-comp";
-            version = self.rev or "dirty";
-
-            src = nix-filter.lib.filter {
-              root = ./.;
-              include = [
-                ./src
-                ./i18n.toml
-                ./Cargo.toml
-                ./Cargo.lock
-                ./resources
-                ./cosmic-comp-config
-              ];
-            };
-
-            nativeBuildInputs = with pkgs; [
-              pkg-config
-              autoPatchelfHook
-              cmake
-            ];
+  outputs = inputs @ {
+    self,
+    crane,
+    ...
```

Comment:
It would now be sane to take all inputs instead of using `inputs.systems` and `inputs.rust-overlay`